app/helpers/patient_query_helper.rb

```diff
@@ -74,11 +75,12 @@ def validate_patients_query(unsanitized_query)
       raise InvalidQueryError.new(:tz_offset, tz_offset) unless tz_offset.to_i.to_s == tz_offset.to_s
 
       query[:filter] = unsanitized_query[:filter].collect do |filter|
-        permitted_filter_params = filter.permit(:value, :numberOption, :dateOption, :relativeOption, :additionalFilterOption, filterOption: {}, value: {})
+        permitted_filter_params = filter.permit(:value, :numberOption, :dateOption, :relativeOption, :additionalFilterOption,
+                                                filterOption: {}, value: filter[:value].is_a?(Array) ? %i[label value] : {})
         {
           filterOption: filter.require(:filterOption).permit(:name, :title, :description, :type, :hasTimestamp, :tooltip,
                                                              options: [], fields: [:name, :title, :type, { options: [] }]),
-          value: permitted_filter_params[:value] || filter.require(:value) || false,
+          value: permitted_filter_params[:value] || false,
```

Comment:
these also need to be updated:

https://github.com/SaraAlert/SaraAlert/blob/master/app/controllers/user_filters_controller.rb#L24

https://github.com/SaraAlert/SaraAlert/blob/master/app/controllers/user_filters_controller.rb#L40

![Screen Shot 2021-08-10 at 1 19 42 PM](https://user-images.githubusercontent.com/9956561/128905433-b0139a4b-f565-4bca-b660-d1c2de81b9ef.png)